spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/README.md

```diff
@@ -0,0 +1,176 @@
+	 $ mvn clean package	 
+	 $ export SPRING_REDIS_HOST=redis.default.svc.cluster.local
+	 $ java -jar spring-cloud-dataflow-admin/target/spring-cloud-dataflow-admin-1.0.0.BUILD-SNAPSHOT.jar --spring.profiles.active=kubernetes 
+	 ```
+
+1. Run Spring Cloud Data Flow Shell:
+
+   ```
+   $ java -jar spring-cloud-dataflow-shell/target/spring-cloud-dataflow-shell-1.0.0.BUILD-SNAPSHOT.jar
+   ```
+
+
+1. Deploy your first stream to Kubernetes
+   
+   ```
+   dataflow:> stream create --name deployment1 --definition "http --server.port=9999 | log"  --deploy
+   ```
+   
+   You can now check the status:
+   
+   ```
+   dataflow:>stream list
+   Stream Name  Stream Definition              Status
+   -----------  -----------------------------  ---------
+   deployment1  http --server.port=9999 | log  deploying
+   ```
+
+   The first deployment may take some time as Kubernetes needs to fetch the base image
+   for running Spring Cloud Stream modules from DockerHub. Once the status is `deployed`
+   you can test the stream.
+   
+   Send a log message to the http endpoint that was created by the stream:
+   
+   ```
+   $ curl -v -XPOST -H"Content-Type: text/plain" http://deployment1-http.default.svc.cluster.local:9999/messages -d "foobar123"
+   ```
+   
+   Check the log output:
+   ```
+	 $ kubectl get pods -l scsm-label=log
+	 NAME                    READY     STATUS    RESTARTS   AGE
+	 deployment1-log-tacrf   1/1       Running   0          7m 
+
+	 $ kubectl logs deployment1-log-tacrf
+   ```
+   
+   The last line should say "foobar123". This means the stream is working correctly.
+   
+	
+	 Also you check the status via Kubernetes `kubectl`:
+	 ```
+	 $ kubectl get service,rc,pods
+	 ```
+	 
+	 You will see a bunch of new resources being deployed.
+	 
+	 
+
+
+### Configuration settings
+
+Internally the Kubernetes Deployer uses the excellent [Kubernetes Client API](https://github.com/fabric8io/kubernetes-client).
+
+If you have a working `kubectl` installation that points to an exsting Kubernetes cluster, the Spring Admin + the Kubernetes deployer will pick up the configuration automatically (in most cases). If not, see the configuration options below.
+
+If you want to use your own base image you can set the following arguments when starting the Spring Admin:
+
+```
+--kubernetes.imagePullSecret=mysecret 
+--kubernetes.moduleLauncherImage=myorg/mycustom-module-launcher-image
+```
+
+The default image is: [https://hub.docker.com/r/springcloud/stream-module-launcher/](springcloud/stream-module-launcher).
+
+
+##### TL;DR 
+
+It will use settings from different sources in the following order of priority:
+
+* System properties
+* Environment variables
+* Kube config file
+* Service account token & mounted CA certificate
+
+System properties are preferred over environment variables. The following system properties & environment variables can be used for configuration:
+
+* `kubernetes.master` / `KUBERNETES_MASTER`
```

Comment:
We're working on supporting common properties for all deployers (like setting the amount of RAM for a module, _etc._) but I see this deployer implementation does not support that at all. Is this by design?

Also, about the props below, are those things that are recognized by other tools of the kubernetes ecosystem (_e.g._ `kubectl` installation as mentioned above). If so, we should think about whether or not using the same `kubernetes.` prefix is a good idea